File: scratch/full_fresh_reset.py
import asyncio
import os
import sys

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from services import database

async def main():
    print("Connecting to DB via services.database...")
    await database.init_db()
    db = database.db_conn
    
    # 1. Reset all users balance to 0
    await db.execute("UPDATE users SET balance = 0")
    print("[SUCCESS] All users balance reset to 0 in database.")

    # 2. Clear all orders
    try:
        await db.execute("DELETE FROM orders")
        print("[SUCCESS] All orders cleared from database.")
    except Exception as e:
        print("Orders table error:", e)

    # 3. Clear balance history
    try:
        await db.execute("DELETE FROM balance_history")
        print("[SUCCESS] Balance history cleared.")
    except Exception as e:
        print("Balance history error:", e)

    # 4. View users
    users = await db.fetch("SELECT telegram_id, username, balance FROM users LIMIT 25")
    print("\n--- ALL USERS IN DB (BALANCES ZERO) ---")
    for u in users:
        print(f"ID: {u['telegram_id']} | Username: @{u['username']} | Balance: {u['balance']} UZS")

    print("\n[COMPLETE] Fresh reset completed successfully!")

if __name__ == "__main__":
    asyncio.run(main())

File: scratch/full_fresh_reset.js
const { Pool } = require('pg');

async function fullFreshReset() {
  const connectionString = process.env.DATABASE_URL || 'postgresql://neondb_owner:npg_whMk3x5XVFTz@ep-aged-voice-ax3rogww-pooler.c-4.us-east-2.aws.neon.tech/neondb?sslmode=require';
  const pool = new Pool({
    connectionString,
    ssl: { rejectUnauthorized: false },
  });

  try {
    console.log('--- STARTING FRESH DATABASE RESET ---');

    // 1. Reset all users balance to 0
    const userRes = await pool.query('UPDATE users SET balance = 0');
    console.log(`✅ Reset balance to 0 for ${userRes.rowCount} users.`);

    // 2. Clear all orders
    try {
      const orderRes = await pool.query('DELETE FROM orders');
      console.log(`✅ Cleared ${orderRes.rowCount} test orders from orders table.`);
    } catch (e) {
      console.log('Orders table clear info:', e.message);
    }

    // 3. Clear balance history / transactions if exist
    try {
      await pool.query('DELETE FROM balance_history');
      console.log('✅ Cleared balance_history table.');
    } catch (e) {}

    try {
      await pool.query('DELETE FROM transactions');
      console.log('✅ Cleared transactions table.');
    } catch (e) {}

    // 4. Verify fresh state
    const users = await pool.query('SELECT telegram_id, username, balance FROM users');
    console.log('\n📊 USERS CURRENT STATE:');
    console.table(users.rows);

    const ordersCount = await pool.query('SELECT COUNT(*) FROM orders');
    console.log(`📦 TOTAL ORDERS COUNT: ${ordersCount.rows[0].count}`);

    console.log('\n🎉 FRESH RESET COMPLETED SUCCESSFULLY!');
  } catch (err) {
    console.error('Error during fresh reset:', err);
  } finally {
    await pool.end();
  }
}

fullFreshReset();
